cfbs/internal_file_management.py

```diff
@@ -155,15 +156,9 @@ def _clone_and_checkout(url, path, treeish):
     sh("git checkout " + treeish, directory=path)
 
 
-def clone_url_repo(repo_url: str):
+def clone_url_repo(repo_url: str, commit: Optional[str] = None):
     assert repo_url.startswith(SUPPORTED_URI_SCHEMES)
-
-    commit = None
-    if "@" in repo_url and (repo_url.rindex("@") > repo_url.rindex(".")):
-        # commit specified in the url
-        repo_url, commit = repo_url.rsplit("@", 1)
-        if not is_a_commit_hash(commit):
-            raise CFBSExitError("'%s' is not a commit reference" % commit)
+    assert "@" not in repo_url or (repo_url.rindex("@") < repo_url.rindex("."))
```

Comment:
Not a big deal, and probably done this way in other places, but to me, using an if looks more readable (close to how I would explain it) rather than the "or" way;

```python
if "@" in repo_url:
  assert repo_url.rindex("@") < repo_url.rindex(".")
```